contrib/epee/include/net/net_helper.h

```diff
@@ -562,11 +553,9 @@ namespace net_utils
 	protected:
 		boost::asio::io_service m_io_service;
 		boost::asio::ip::tcp::socket m_socket;
-		int m_connect_timeout;
-		int m_reciev_timeout;
 		bool m_initialized;
 		bool m_connected;
-		boost::asio::deadline_timer m_deadline;
+		boost::asio::steady_timer m_deadline;
```

Comment:
Going from `boost::posix_time` to `std::chrono` meant a different ASIO timer has to be used. Internally ASIO forwards to the same OS specific calls, but the original `asio::deadline_timer` doesn't support `sd::chrono`.

As for moving to `std::chrono`...

Previously the functions for invoking remote commands were using `unsigned` types for specifying timeouts. There was no indication in the comments, name, or type for the unit of time. Since the usage of the function was changing (to use relative path instead of specifying full path), I decided to take the opportunity to update the type for the timeout parameter, so that the unit of time was explicit. The `std::chrono` library is the C++11 way of dealing with time. Benefits:
- Durations (`std::chrono::seconds`, etc) and specific points in time (`std::chrono::steady_clock::time_point`, etc) are fast - just wrappers around a single integer.
- Arithmetic with different units of time are automatically scaled to smallest unit in expression
- Assigning to a unit of time that would cause precision loss is a compiler failure unless `duration_cast` is used (I.e. `std::chrono::seconds timeout = std::chrono::milliseconds(10);` is a compile failure).
- Durations are fully `constexpr` enabled, and values known at compile are constant folded like normal integers (i.e. `std::chrono::milliseconds foo = std::chrono::hours(10) + std::chrono::seconds(33));`) can be computed/folded at compile-time
- Durations are not implicitly constructible from integers. This forces callees to specify a unit of time, which means changing the unit of time in the function signature never alters the behavior.

I think `std::chrono` should be preferred because it is `constexpr` enabled, and the duration types avoid the base class `posix::time_duration` (which has an "implemention defined" unit of time and strange constructor).

The codebase is already using `std::chrono` in a few select areas.

Thoughts ... ?